crates/parsing/src/grammar/snapshots/parsing__grammar__tests__expect_parse@constrained.input.snap

```diff
@@ -5,29 +5,40 @@ input_file: crates/parsing/src/grammar/inputs/passing/type/constrained.input
 ---
 Input: A => B => a
 
-Start { kind: ConstrainedType }
+Start { kind: TypeOperatorChain }
```

Comment:
Since `=>` is built-in and cannot be re-defined, I think this should be parsed as `ConstrainedType`. I haven't implemented desugaring for `TypeOperatorChain` in the analyzer yet but `ConstrainedType` and `ArrowType` should be kept as-is.